go/vt/vtgate/plugin_mysql_server.go

```diff
@@ -621,18 +623,28 @@ func newMysqlUnixSocket(address string, authServer mysql.AuthServer, handler mys
 }
 
 func (srv *mysqlServer) shutdownMysqlProtocolAndDrain() {
-	if srv.tcpListener != nil {
-		srv.tcpListener.Shutdown()
-		srv.tcpListener = nil
-	}
-	if srv.unixListener != nil {
-		srv.unixListener.Shutdown()
-		srv.unixListener = nil
-	}
 	if srv.sigChan != nil {
 		signal.Stop(srv.sigChan)
 	}
 
+	if mysqlDrainOnTerm {
+		stopListener(srv.unixListener, false)
+		stopListener(srv.tcpListener, false)
```

Comment:
the `stopListener` method does not reset `srv.tcpListener` and `srv.unixListener` to `nil`. Mostly not an issue, just pointing out.

Reply:
Fixed via c630c19c98